packages/gradient-block/src/GradientBlock.tsx

```diff
@@ -0,0 +1,147 @@
+import { BlockProps } from '@frontify/guideline-blocks-settings';
+import { Divider } from '@frontify/fondue';
+import 'tailwindcss/tailwind.css';
+import { GradientColor, Settings, gradientHeightValues, gradientOrientationValues } from './types';
+import { DEFAULT_GRADIENT_COLORS, DEFAULT_HEIGHT_VALUE, DEFAULT_ORIENTATION_VALUE } from './constants';
+import { AddColorButton, ColorPicker, ColorTooltip, CssValueDisplay, SquareBadgesRow } from './components';
+import { toHexString } from '@frontify/guideline-blocks-shared';
+
+export const GradientBlock = ({ appBridge }: BlockProps) => {
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const isEditing = useEditorState(appBridge);
+    const {
+        gradientColors,
+        isOrientationCustom,
+        orientationCustom,
+        orientationSimple,
+        isHeightCustom,
+        heightCustom,
+        heightSimple,
+        displayCss,
+    } = blockSettings;
+    const gradientBlockRef = useRef<HTMLDivElement>(null);
+    const [addButtonPositionLeft, setAddButtonPositionLeft] = useState(0);
+    const [currentlyEditingPosition, setCurrentlyEditingPosition] = useState(0);
+    const [showAddButton, setShowAddButton] = useState(false);
+    const [showColorModal, setShowColorModal] = useState(false);
+    const initialColors = !gradientColors || !gradientColors?.length ? DEFAULT_GRADIENT_COLORS : gradientColors;
+    const [colors, setColors] = useState<GradientColor[]>(initialColors);
+
+    const gradientOrientation = isOrientationCustom
+        ? orientationCustom
+        : gradientOrientationValues[orientationSimple ?? DEFAULT_ORIENTATION_VALUE];
+
+    const gradientBlockHeight = isHeightCustom
+        ? heightCustom
+        : gradientHeightValues[heightSimple ?? DEFAULT_HEIGHT_VALUE];
+
+    const parseGradientColorsToCss = () => {
+        let colorsAsString = '';
+        for (const color of colors.sort((a, b) => a.position - b.position)) {
+            colorsAsString += `, ${toHexString(color.color)} ${color.position}%`;
+        }
+
+        return `linear-gradient(${gradientOrientation}deg${colorsAsString})`;
+    };
+
+    const handleMouseMove = (event: MouseEvent<HTMLDivElement>) => {
+        if (showColorModal) {
+            return;
+        }
+        const rect = event.currentTarget.getBoundingClientRect();
+        const { clientX } = event.nativeEvent;
+        const relativeMouseX = clientX - rect.x;
+        setAddButtonPositionLeft(relativeMouseX - 16 / 2);
+        setShowAddButton(true);
+    };
+
+    useEffect(() => {
+        if (!gradientColors || !gradientColors?.length) {
+            setBlockSettings({
+                gradientColors: DEFAULT_GRADIENT_COLORS,
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [gradientColors]);
+
+    useEffect(() => {
+        setBlockSettings({
+            gradientColors: colors,
```

Comment:
What is the purpose behind having a separate `colors` state and not directly setting the `gradientColors` in the blockSettings?

Reply:
Why is gradientOrientation needed in the dependency array?

Reply:
You are completely right this state was "useless" and so was the useEffects in the end, both are removed

Reply:
But i need to add an useEffect and useState for isRefReady otherwise the squareBadges wouldn't render within cypress